File: lc-0038/solution.cpp
class Solution {
public:
    string countAndSay(int n) {
        string s = "1";
        for (int i = 1; i < n; i++) {
            s = say(s);
        }
        return s;
    }

    string say(const string& s) {
        string r;
        char c = s[0];
        int n = 1;
        for (int i = 1; i < s.size(); i++) {
            if (s[i] == c) {
                n++;
            } else {
                r += to_string(n) + string(1, c);
                c = s[i];
                n = 1;
            }
        }
        r += to_string(n) + string(1, c);
        return r;
    }
};

File: lc-0038/solution.rs
impl Solution {
    pub fn count_and_say(n: i32) -> String {
        let say = |s: String| {
            let mut r = String::new();
            let mut c = s.as_bytes()[0];
            let mut n = 1;
            for i in 1..s.len() {
                let ch = s.as_bytes()[i];
                if ch == c {
                    n += 1;
                } else {
                    r += &n.to_string();
                    r.push(c as char);
                    c = ch;
                    n = 1;
                }
            }
            r.push_str(&n.to_string());
            r.push(c.into());
            r
        };

        let mut s = "1".to_string();
        for i in 1..n {
            s = say(s);
        }
        s
    }
}
